Services/Init/classes/class.ilInitialisation.php

```diff
@@ -37,17 +37,17 @@ protected static function removeUnsafeCharacters()
 		// We do not need this characters in any case, so it is
 		// feasible to filter them everytime. POST parameters
 		// need attention through ilUtil::stripSlashes() and similar functions)
-		self::recursivelyRemoveUnsafeCharacters($_GET);
+		$_GET = self::recursivelyRemoveUnsafeCharacters($_GET);
 	}
 
-	protected static function recursivelyRemoveUnsafeCharacters(&$var) {
+	protected static function recursivelyRemoveUnsafeCharacters($var) {
 		if (is_array($var)) {
+			$mod = [];
```

Comment:
This is ok for 5.2. But can not be merged to 5.1 due to short array syntax.

Reply:
You are right, short array syntax came with PHP 5.4. ILIAS 5.1 supports PHP 5.3.....